flink-java/src/main/java/org/apache/flink/api/java/typeutils/TypeExtractor.java

```diff
@@ -340,6 +340,15 @@ protected TypeExtractor() {
 		}
 		return ti;
 	}
+
+	@SuppressWarnings("unchecked")
+	public static <OUT> TypeInformation<OUT> createTypeInfo(Class<?> baseClass, Class<?> clazz, int returnParamPos, Object instance) {
```

Comment:
I would maybe move the `instance` parameter to the front of the parameter list. Then it's similar to `getUnaryOperatorReturnType`.